File: 15.py
#!/usr/bin/env python

"""This script solves the Project Euler problem "Lattice paths". The problem
is: How many such routes are there through a 20x20 grid?
"""

from __future__ import division
import argparse
import math


def main(args):
    """Lattice paths"""

    size = args.grid_size
    routes = math.factorial(size + size) / math.pow(math.factorial(size), 2)

    print(int(routes))

if __name__ == '__main__':
    parser = argparse.ArgumentParser(description='Lattice paths')
    parser.add_argument(
        'grid_size', metavar='GRID_SIZE', type=int, default=20, nargs='?',
        help='The size of the grid (INT x INT)')
    args = parser.parse_args()

    main(args)

File: 15.js
#!/usr/bin/env node

// Lattice paths

var program = require('commander');

program
    .version('0.1.0')
    .description('Lattice paths')
    .option('-g, --grid_size <int>', 'The size of the grid', Number, 20)
    .parse(process.argv);

var size = program.grid_size;
console.log(Math.round(factorial(size + size) / Math.pow(factorial(size), 2)));

function factorial(number) {
    var factorial = 1;

    while (number > 1) {
        factorial *= number;
        number--;
    }

    return factorial;
}
